les/fetcher.go

```diff
@@ -749,19 +759,20 @@ func (f *lightFetcher) checkSyncedHeaders(p *peer) {
 }
 
 // lastValidTrieNode return last approved treeNode and a list of unapproved hashes
-func (f *lightFetcher) lastValidTrieNode(p *peer) (*fetcherTreeNode, []common.Hash) {
+func (f *lightFetcher) lastValidTrieNode(p *peer) (*types.Header, []common.Hash) {
 	unapprovedHashes := make([]common.Hash, 0)
 
-	fp := f.peers[p]
-	last := fp.lastAnnounced
-	for last != nil {
-		if f.isTrustedHash(last.hash) {
+	current := f.chain.CurrentHeader()
+	for current != nil || current != f.lastTrustedHeader {
```

Comment:
Here is where I meant the comparison of the `lastTrustedHeader` set async in a goroutine.